### ISSUE TITLE
output debug messages to standard error stream

### ISSUE THREAD

**Issue by petrovr** [open-quantum-safe/oqs-provider#594]
Currently macros OQS_*_PRINTF? call function printf(..). Debug message is mixed with regular program output and this break functionality. For instance if output of one program is redirected as input to another one.

Technically this is request is to replace "printf" with "fprintf(stderr," and if needed to include stdio.h header.

---- 

In addition oqsprov/oqsprov.c also calls printf once.

**Comment by baentsch** [comment]
Agree, this is "sub optimal" when using a Debug build... But I do not see it as a bug but as a nuisance (on my list for improvements in #375... As I don't know yet when/if I or anyone else in the community gets around to doing that, though, please feel free to submit a PR to change this if this would improve things for you).
